src/components/FeatureCards/FeatureCards.tsx

```diff
@@ -5,36 +5,67 @@ import { FeatureCard } from 'src/components';
 import { useFeatureCards } from 'src/hooks';
 import { useSettingsStore } from 'src/stores';
 import type { FeatureCardData } from 'src/types';
+import { deepMerge, removeEmpty } from 'src/utils';
 import { shallow } from 'zustand/shallow';
 import { FeatureCardsContainer } from '.';
 
 export const FeatureCards = () => {
   const [featureCards, setFeatureCards] = useState<FeatureCardData[]>([]);
-  const [disabledFeatureCards, welcomeScreenClosed] = useSettingsStore(
-    (state) => [state.disabledFeatureCards, state.welcomeScreenClosed],
-    shallow,
-  );
+  const [disabledFeatureCards, welcomeScreenClosed, languageMode] =
+    useSettingsStore(
+      (state) => [
+        state.disabledFeatureCards,
+        state.welcomeScreenClosed,
+        state.languageMode,
+      ],
+      shallow,
+    );
+  // get english response
+  const { featureCards: defaultFeatureCards, isSuccess } = useFeatureCards();
+  // get translated response
+  const { featureCards: translatedFeatureCards } = useFeatureCards({
+    translation: true,
+  });
+
+  // remove null´s (missing entries) from translated object
+  const cleanedTranslation = useMemo(() => {
+    const cleanedTranslation = translatedFeatureCards?.map((el) =>
+      removeEmpty(el),
+    );
+    // merge
+    const filteredResponse = defaultFeatureCards?.map((el) => {
+      const translatedItem = cleanedTranslation?.find(
+        (foundItem) =>
+          foundItem?.attributes?.DisplayConditions.id ===
+          el.attributes?.DisplayConditions.id,
+      );
+      if (translatedItem) {
+        return deepMerge(el, translatedItem);
+      }
+      return el;
+    });
 
-  const { featureCards: data, isSuccess } = useFeatureCards();
-  const featureCardsFetched = useMemo(() => {
-    if (Array.isArray(data) && !!data.length) {
-      return data?.filter(
+    if (Array.isArray(filteredResponse) && !!filteredResponse.length) {
+      return filteredResponse?.filter(
         (el, index) =>
           isSuccess &&
           el.attributes.DisplayConditions &&
           !disabledFeatureCards.includes(el.attributes.DisplayConditions?.id),
       );
     }
-    // trigger featureCardsFetched-filtering only once
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [data, isSuccess]);
+  }, [
+    defaultFeatureCards,
+    disabledFeatureCards,
+    isSuccess,
+    translatedFeatureCards,
+  ]);
 
   useEffect(() => {
-    if (Array.isArray(featureCardsFetched)) {
-      !!featureCardsFetched.length &&
-        setFeatureCards(featureCardsFetched?.slice(0, 4));
+    if (Array.isArray(cleanedTranslation)) {
+      !!cleanedTranslation.length &&
+        setFeatureCards(cleanedTranslation?.slice(0, 4));
     }
-  }, [featureCardsFetched]);
+  }, [cleanedTranslation]);
```

Comment:
I think we can move this logic to useMemo as well. There is no need for additional effect and re-render just to slice an array 🙂